controllers/choose_removals.go

```diff
@@ -70,6 +70,9 @@ func (c chooseRemovals) reconcile(ctx context.Context, r *FoundationDBClusterRec
 
 	localityMap := make(map[string]locality.Info)
 	for _, process := range status.Cluster.Processes {
+		if cluster.Spec.DataCenter != "" && cluster.Spec.DataCenter != process.Locality[fdbv1beta2.FDBLocalityDCIDKey] {
```

Comment:
It might be worth moving this into a dedicated method instead of repeating the same code piece multiple times.